pkg/apis/core/validation/validation.go

```diff
@@ -4106,12 +4160,51 @@ func ValidatePodUpdate(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 		return allErrs
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.InPlacePodVerticalScaling) {
```

Comment:
In general we want to avoid checking gates in validation.  If it is required to produce a correct error, it's sometimes OK, but very limited in scope (e.g. you would not want to return "QOS is immutable" when the gate is off, you'd rather return "pod resources are immutable").

My point being - can we express this in a less conditional way, so the almost all of the validation applies regardless of the gate?

Reply:
We do return "pod updates may not change fields other than XYZ" where  XYZ is allowed list, which containers "resources (cpu/memory)" when gate is on.

This retains the current behavior without the long hard-coded string. Of-course I'm now counting on status persistence to help get rid of that "QOS is immutable" block entirely (hopefully before beta)

The current code is uglier than I'd like compared to the previous where I add to the updatableSpecFields error when FG is on, but this cut stack usage and will look so much cleaner upon GA.